src/handlers/deeplinks.ts

```diff
@@ -337,3 +354,71 @@ function handleWalletConnect(uri?: string, connector?: string) {
     store.dispatch(walletConnectSetPendingRedirect());
   }
 }
+
+const isValidAsset = (s: string | undefined): s is UniqueId => {
+  if (!s) return false;
+  const [address, chainId] = s.split('_');
+  return supportedSwapChainIds.includes(parseInt(chainId, 10)) && (address === 'eth' || address.length === 42);
+};
```

Comment:
what do you think about just deriving the `address` and `chainId` in this function? You're basically checking the same thing after. This function could:
1. check if address and chainId are present and valid according to our `supportedSwapChainIds`, and then return the `address` and `chainId` to be used later in the `querySearchAsset ` function.